Initializing services with a Custom URL crashes when no logger has been provided

I've tried to use a custom URL (#125) for teams notification using Kured v1.7.0, but got the error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x14747fb]

goroutine 26 [running]:
github.com/containrrr/shoutrrr/pkg/router.(*ServiceRouter).initService(0x2330de0, 0x7ffc102cdcc0, 0xd3, 0x30, 0xc000042870, 0x30, 0x1)
	/home/runner/go/pkg/mod/github.com/containrrr/shoutrrr@v0.4.4/pkg/router/router.go:189 +0x17b
github.com/containrrr/shoutrrr/pkg/router.(*ServiceRouter).Locate(...)
	/home/runner/go/pkg/mod/github.com/containrrr/shoutrrr@v0.4.4/pkg/router/router.go:238
github.com/containrrr/shoutrrr.Send(0x7ffc102cdcc0, 0xd3, 0xc000042870, 0x30, 0x1, 0xc000042870)
	/home/runner/go/pkg/mod/github.com/containrrr/shoutrrr@v0.4.4/shoutrrr.go:17 +0x45
main.drain(0xc000015080, 0xc0006f8000)
	/home/runner/work/kured/kured/cmd/kured/main.go:351 +0x6b7
main.rebootAsRequired(0xc00004002e, 0xc, 0xc0002061e0, 0x5, 0x6, 0xc000206180, 0x6, 0x6, 0xc000326240, 0x22ecb25c000, ...)
	/home/runner/work/kured/kured/cmd/kured/main.go:569 +0x676
created by main.root
	/home/runner/work/kured/kured/cmd/kured/main.go:644 +0xc05
```

The custom URL used:
`--notify-url=teams+https://<tName>.webhook.office.com/webhookb2/group@tenant/IncomingWebhook//altid/groupowner`

The problem seems to be here:
https://github.com/containrrr/shoutrrr/blob/23170b836d31a2231011472b6e8a91bf98bf6b26/pkg/router/router.go#L189

Maybe router.logger is not initialized when kured calls it?

--- a/pkg/router/router.go
+++ b/pkg/router/router.go
@@ -186,7 +186,7 @@ func (router *ServiceRouter) initService(rawURL string) (t.Service, error) {
 	}
 
 	if configURL.Scheme != scheme {
-		router.logger.Println("Got custom URL:", configURL.String())
+		router.log("Got custom URL:", configURL.String())
 		customURLService, ok := service.(t.CustomURLService)
 		if !ok {
 			return nil, fmt.Errorf("custom URLs are not supported by '%s' service", scheme)
@@ -195,7 +195,7 @@ func (router *ServiceRouter) initService(rawURL string) (t.Service, error) {
 		if err != nil {
 			return nil, err
 		}
-		router.logger.Println("Converted service URL:", configURL.String())
+		router.log("Converted service URL:", configURL.String())
 	}
 
 	err = service.Initialize(configURL, router.logger)
@@ -237,4 +237,11 @@ func (router *ServiceRouter) ListServices() []string {
 func (router *ServiceRouter) Locate(rawURL string) (t.Service, error) {
 	service, err := router.initService(rawURL)
 	return service, err
+}
+
+func (router *ServiceRouter) log(v ...interface{}) {
+	if router.logger == nil {
+		return
+	}
+	router.logger.Println(v...)
 }

--- a/pkg/router/router_suite_test.go
+++ b/pkg/router/router_suite_test.go
@@ -17,6 +17,10 @@ func TestRouter(t *testing.T) {
 
 var sr ServiceRouter
 
+const (
+	mockCustomURL = "teams+https://publicservice.info/webhook/11111111-4444-4444-8444-cccccccccccc@22222222-4444-4444-8444-cccccccccccc/IncomingWebhook/33333333012222222222333333333344/44444444-4444-4444-8444-cccccccccccc"
+)
+
 var _ = Describe("the router suite", func() {
 	BeforeEach(func() {
 		sr = ServiceRouter{
@@ -84,7 +88,7 @@ var _ = Describe("the router suite", func() {
 			Expect(service).To(BeNil())
 		})
 		It("should successfully init a service that does support it", func() {
-			service, err := sr.initService("teams+https://publicservice.info/webhook/11111111-4444-4444-8444-cccccccccccc@22222222-4444-4444-8444-cccccccccccc/IncomingWebhook/33333333012222222222333333333344/44444444-4444-4444-8444-cccccccccccc")
+			service, err := sr.initService(mockCustomURL)
 			Expect(err).NotTo(HaveOccurred())
 			Expect(service).NotTo(BeNil())
 		})
@@ -111,6 +115,13 @@ var _ = Describe("the router suite", func() {
 				defer sr.Flush(nil)
 				sr.Enqueue("message")
 			})
+		})
+	})
+	When("router has not been provided a logger", func() {
+		It("should not crash when trying to log", func() {
+			router := ServiceRouter{}
+			_, err := router.initService(mockCustomURL)
+			Expect(err).NotTo(HaveOccurred())
 		})
 	})
 })

